R/add_to_view.R

```diff
@@ -0,0 +1,58 @@
+
+#' add function call to view
+#' 
+#' @param fun.name
+#' @param \dots
+#' @examples 
+#' gs <- gsplot() %>% 
+#'          points(x=1:5, y=1:5, xlim=c(0,10), ylim=c(0,10), 
+#'                callouts(labels=c(rep(NA, 4), "oh")), 
+#'                error_bar(y.high=1))
+#' gs
+#' 
+#' gsplot:::call_arguments('points', x=2:6, y=2:6, ylim=c(-1, 11))
+#' gsplot:::call_arguments('points', x=1:5, y=1:5, xlim=c(0,10), ylim=c(0,10), 
+#'                callouts(labels=c(rep(NA, 4), "oh")))
+#' @keywords internal
+call_arguments <- function(fun.name, ...){
+  dots <- separate_args(...)
+  
+  norm.args <- normal_arguments(fun.name, dots$args)
+  embed.args <- embedded_arguments(fun.name, dots$e.fun, parent.args=norm.args[[fun.name]])
+  view.fun.args <- append(norm.args, embed.args)
+  return(view.fun.args)
+}
+
+#' get the arguments that go into the function call, stripping out others and adding config defaults
+#' 
+#' @param fun.name
+#' @param .dots lazy_dots
+#' @keywords internal
+normal_arguments <- function(fun.name, .dots){
+
+  fun.defaults <- function_defaults(fun.name)
+  
+  if (!is.null(.dots))
+    arguments = set_args(fun.name, lazy_eval(.dots), package=fun.defaults$package)
```

Comment:
@jread-usgs I am getting an error at this line when I run the example in the gsplotIntro.Rmd. The error is only for abline, works fine when I comment it out. This only happens when abline has `legend.name` defined. 

```r
demoPlot <- gsplot() %>%
  points(y=c(3,1,2), x=1:3, xlim=c(0,NA),ylim=c(0,NA),
         col="blue", pch=18, legend.name="Points", xlab="Index") %>%
  lines(c(3,4,3), c(2,4,6), legend.name="Lines", ylab="Data") %>%
  abline(b=1, a=0, legend.name="1:1")

Error in asNamespace(ns) : not a namespace
```